tlsr9/CMakeLists.txt

```diff
@@ -50,17 +59,27 @@ if(CONFIG_SOC_FLASH_TELINK_B9X OR CONFIG_USB_TELINK_B9X)
 		drivers/${SOC}/stimer.c
 	)
 endif()
+if(CONFIG_SOC_FLASH_TELINK_TLX OR CONFIG_USB_TELINK_TLX)
+	zephyr_library_sources(
+		drivers/${SOC}/flash.c
+	)
+endif()
 
+# GPIO driver reference sources
 zephyr_library_sources_ifdef(CONFIG_GPIO_TELINK_B9X drivers/${SOC}/gpio.c)
+zephyr_library_sources_ifdef(CONFIG_GPIO_TELINK_TLX drivers/${SOC}/gpio.c)
 
 # PWM driver reference sources
 zephyr_library_sources_ifdef(CONFIG_PWM_TELINK_B9X drivers/${SOC}/pwm.c)
+zephyr_library_sources_ifdef(CONFIG_PWM_TELINK_TLX drivers/${SOC}/pwm.c)
 
 # SPI driver reference sources
 zephyr_library_sources_ifdef(CONFIG_SPI_TELINK_B9X drivers/${SOC}/gpio.c)
+zephyr_library_sources_ifdef(CONFIG_SPI_TELINK_TLX drivers/${SOC}/gpio.c)
 
 # I2C driver reference sources
 zephyr_library_sources_ifdef(CONFIG_I2C_TELINK_B9X drivers/${SOC}/i2c.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_TELINK_TLX drivers/${SOC}/i2c.c)
```

Comment:
Same here and in all file
investigate if better to change or leave as it is, from one side in some place we save line of code and make code simple and in others it will change two line into tree
Will be good to have one common solution for all includes in this Make file
for example 
if(CONFIG_GPIO_TELINK_B9X OR CONFIG_GPIO_TELINK_TLX)
	zephyr_library_sources(drivers/${SOC}/gpio.c)
endif()